File: 70. Climbing Stairs/Approach03.py
from typing import *
import collections

class Solution:
    def climbStairs(self, n: int) -> int:
        n1, n2 = 1, 2

        if(n==1 or n==2):
            return n
        
        for i in range(3,n+1):
            n2 = n1 + n2
            n1 = n2 - n1

        return n2

File: 70. Climbing Stairs/Approach03.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int climbStairs(int n) {
        int n1 = 1;
        int n2 = 2;
        if(n==1 or n==2){return n;}
        for(int i=3; i<=n; i++){
            n2 = n1+n2;
            n1 = n2 - n1;
        }
        return n2;
    }
};